src/store/reducers/raducer.js

```diff
@@ -0,0 +1,63 @@
+import {FETCH_MOVIES_BEGIN, FETCH_MOVIES_FAILURE, FETCH_MOVIES_SUCCESS, MOVIE_SELECTED} from '../actions/movies';
+
+const initialState = {
+    data: [],
+    loading: false,
+    error: null,
+    isProductSelected: false,
+    selectedProduct: {},
+    value: '',
+    searchBy: 'title',
+    sortBy: 'vote_average'
+};
+export const reducer = (state = initialState, action) => {
+    switch (action.type) {
+        case FETCH_MOVIES_BEGIN:
+            return {
+                ...state,
+                loading: true,
+                error: null
+            };
+
+        case FETCH_MOVIES_SUCCESS:
+            return {
```

Comment:
we can extract this logic to separate e.g. `reduceFetchMovieSuccess(state, action)`to keep the reducer itself readable